src/app/pages/Project.tsx

```diff
@@ -1,51 +1,98 @@
 import React, {useState, useEffect} from 'react';
 import {
+  fetchProject,
   createDocument,
   deleteDocument,
   fetchDocumentsProject,
+  fetchDocumentsAndMultiPacksProject,
+  createCrossDoc,
+  deleteCrossDoc,
 } from '../lib/api';
 import {Link, useHistory} from 'react-router-dom';
 import DropUpload from '../components/dropUpload';
 import {FileWithPath} from 'react-dropzone';
 
 function Docs() {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [projectInfo, setProjectInfo] = useState<any>();
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const [docs, setDocs] = useState<any[]>([]);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [crossdocs, setCrossDocs] = useState<any[]>();
 
   const history = useHistory();
 
   useEffect(() => {
-    updateDocs().catch(() => {
+    getProjectInfo().catch(() => {
       history.push('/login');
     });
   }, [history]);
 
-  function updateDocs() {
-    const project_id = window.location.pathname.split('/').pop()!;
+  useEffect(() => {
+    updateDocs();
+  }, [projectInfo]);
```

Comment:
instead of use useEffect here, why not just call `updateDocs();` after `setProjectInfo(info);`?

Reply:
`updatedocs` must be called after `setProjectInfo` is done because it calls different api if the project type is "cross doc"

Reply:
with `updatedocs(projectInfo)`, I think we don't need useEffect